src/test/java/no/unit/bibs/contents/document/DocumentServiceTest.java

```diff
@@ -0,0 +1,118 @@
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import nva.commons.apigateway.exceptions.BadRequestException;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+class DocumentServiceTest extends DocumentTestBase {
+
+    @BeforeEach
+    void setUp() throws JsonProcessingException, BadRequestException {
+        init();
+    }
+
+    @Test
+    void isValidReturnsFalseWhenIsbnIsBlank() {
+        var document = DocumentDao.builder()
+            .isbn("")
+            .title("Test Title")
+            .author("Test Author")
+            .build()
+            .toDto();
+        assertThrows(BadRequestException.class, () -> documentService.validateDocument(document));
+    }
+
+    @Test
+    void isValidReturnsFalseWhenTitleIsBlank() {
+        var document = DocumentDao.builder()
+            .isbn("978-3-16-148410-0")
+            .title("")
+            .author("Test Author")
+            .build()
+            .toDto();
+        assertThrows(BadRequestException.class, () -> documentService.validateDocument(document));
+    }
+
+    @Test
+    void isValidReturnsFalseWhenAuthorIsBlank() {
+        var document = DocumentDao.builder()
+            .isbn("978-3-16-148410-0")
+            .title("Test Title")
+            .author("")
+            .build()
+            .toDto();
+        assertThrows(BadRequestException.class, () -> documentService.validateDocument(document));
+    }
+
+    @Test
+    void isValidReturnsTrueWhenAllFieldsAreValid() throws BadRequestException {
+        var document = DocumentDao.builder()
+            .isbn("978-3-16-148410-0")
+            .source("http://example.com/source")
+            .descriptionShort("Kort beskrivelse")
+            .imageSmall("http://example.com/small.jpg")
+            .tableOfContents("Innholdsfortegnelse")
+            .build()
+            .toDto();
+        assertTrue(document.isValid());
+    }
+
+    @Test
+    void isValidReturnsTrueWhenDescriptionMissing() throws BadRequestException {
+        var document = DocumentDao.builder()
+            .isbn("978-3-16-148410-0")
+            .source("http://example.com/source")
+            .imageSmall("http://example.com/small.jpg")
+            .build()
+            .toDto();
+        assertTrue(document.isValid());
+
+    }
+
+    @Test
+    void isValidReturnsTrueWhenImageMissing() throws BadRequestException {
+        var document = DocumentDao.builder()
+            .isbn("978-3-16-148410-0")
+            .source("http://example.com/source")
+            .descriptionShort("Kort beskrivelse")
+            .build()
+            .toDto();
+        assertTrue(document.isValid());
+    }
+
+
+    @Test
+    void isValidReturnsFalseWhenBothMissing() throws BadRequestException {
```

Comment:
so here, since anyDescription is blank, meaning the document does not contain any data other than metadata like isbn and source, the document is not valid. maybe you should name the test to reflect that